src/Lorentz/Contracts/GameDAO.hs

```diff
@@ -0,0 +1,202 @@
+  ( GameDaoProposalMetadata (..)
+  , Storage
+  , Parameter
+  , ProposalType (..)
+  , BalanceChange (..)
+  , ItemChange (..)
+  , HeroChange (..)
+  , NewContents
+  , NewContent (..)
+
+  , gameDaoContract
+  , emptyStorage
+  , config
+  ) where
+
+import Universum hiding ((>>), drop, swap)
+import Michelson.Runtime.GState (genesisAddress)
+import Lorentz
+import qualified Lorentz.Contracts.BaseDAO as DAO
+import qualified Lorentz.Contracts.BaseDAO.Types as DAO
+import Util.Markdown
+
+data GameDaoProposalMetadata = GameDaoProposalMetadata
+  { pmProposalType :: ProposalType
+  , pmProposalDescription :: MText
+  }
+  deriving stock (Generic)
+  deriving anyclass IsoValue
+
+instance HasAnnotation GameDaoProposalMetadata where
+  annOptions = DAO.baseDaoAnnOptions
+
+data ProposalType = BalanceType BalanceChange | NewType NewContents
+  deriving stock (Generic)
+  deriving anyclass (IsoValue)
+
+instance HasAnnotation ProposalType where
+  annOptions = defaultAnnOptions { fieldAnnModifier = toSnake }
+
+instance TypeHasDoc ProposalType where
+  typeDocMdDescription = "There are 2 types of proposal that Game DAO supports. \
+    \'balance_type' is for small update to existing mechanic, items and heroes. \
+    \'new_type' is for proposing new contents all together."
+
+data BalanceChange = BalanceChange
+  { bcItemChanges :: [ItemChange]
+  , bcHeroChanges :: [HeroChange]
+  }
+  deriving stock (Generic)
+  deriving anyclass (IsoValue)
+
+instance HasAnnotation BalanceChange where
+  annOptions = DAO.baseDaoAnnOptions
+
+instance TypeHasDoc BalanceChange where
+  typeDocMdDescription = "Describe the update of all items and heroes in the proposal."
+
+data ItemChange = ItemChange
+  { icItemName :: MText
+  , icChangelogs :: [MText]
+  }
+  deriving stock (Generic)
+  deriving anyclass (IsoValue)
+
+instance TypeHasDoc ItemChange where
+  typeDocMdDescription = "Describe the update of a specific item."
+
+instance HasAnnotation ItemChange where
+  annOptions = DAO.baseDaoAnnOptions
+
+data HeroChange = HeroChange
+  { hcHeroName :: MText
+  , hcChangelogs :: [MText]
+  }
+  deriving stock (Generic)
+  deriving anyclass (IsoValue)
+
+instance TypeHasDoc HeroChange where
+  typeDocMdDescription = "Describe the update of a specific hero."
+
+instance HasAnnotation HeroChange where
+  annOptions = DAO.baseDaoAnnOptions
+
+type NewContents = [NewContent]
+
+data NewContent = NewContent
+  { ncContentName :: MText
+  , ncContentDescription :: [MText]
+  }
+  deriving stock (Generic)
+  deriving anyclass (IsoValue)
+
+instance TypeHasDoc NewContent where
+  typeDocMdDescription = "Describe a new content. The content should have a name and detail description."
+
+instance HasAnnotation NewContent where
+  annOptions = DAO.baseDaoAnnOptions
+
+instance TypeHasDoc GameDaoProposalMetadata where
+  typeDocMdDescription = "GameDAO's metadata. This fields affect how proposal got accepted and how many tokens will be frozen"
```

Comment:
This line looks too long btw, could you please wrap it? Same for ` instance TypeHasDoc NewContent`.